packers/zkp.go

```diff
@@ -117,8 +118,47 @@ func (p *ZKPPacker) Unpack(envelope []byte) (*iden3comm.BasicMessage, error) {
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+
+	// check that sender of the message is presented in proof
+	err = verifySender(token, msg)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
 	return &msg, err
 }
+func verifySender(token *jwz.Token, msg iden3comm.BasicMessage) error {
+
+	bytePubsig, err := json.Marshal(token.ZkProof.PubSignals)
+	if err != nil {
+		return err
+	}
+
+	var (
```

Comment:
since you announced only one variable please, refactor to
var userID *big.Int

Reply:
fixed